File: frontend_old/src/services/therapist.ts
import api from './api';

export interface Therapist {
    id: string;
    first_name: string;
    last_name: string;
    email: string;
    phone?: string;
    license_number: string;
    license_state: string;
    years_of_experience?: number;
    assessment_count: number;
    rating: number;
    is_active: boolean;
    created_at: string;
    updated_at: string;
}

export interface CreateTherapistData {
    first_name: string;
    last_name: string;
    email: string;
    phone?: string;
    license_number: string;
    license_state: string;
    years_of_experience?: number;
}

const TherapistService = {
    getTherapists: async () => {
        const response = await api.get<Therapist[]>('/therapists');
        return response.data;
    },

    getTherapist: async (id: string) => {
        const response = await api.get<Therapist>(`/therapists/${id}`);
        return response.data;
    },

    createTherapist: async (data: CreateTherapistData) => {
        const response = await api.post<Therapist>('/therapists', data);
        return response.data;
    },

    updateTherapist: async (id: string, data: Partial<CreateTherapistData>) => {
        const response = await api.put<Therapist>(`/therapists/${id}`, data);
        return response.data;
    },

    deleteTherapist: async (id: string) => {
        const response = await api.delete<Therapist>(`/therapists/${id}`);
        return response.data;
    },

    updateRating: async (id: string, rating: number) => {
        const response = await api.post<Therapist>(`/therapists/${id}/update-rating`, { rating });
        return response.data;
    }
};

export default TherapistService;

File: api/schemas/therapist.py
from typing import Optional
from pydantic import BaseModel, EmailStr, Field
from datetime import datetime
from uuid import UUID

class TherapistBase(BaseModel):
    first_name: str
    last_name: str
    email: EmailStr
    phone: Optional[str] = None
    license_number: str
    license_state: str
    years_of_experience: Optional[int] = Field(None, ge=0)
    rating: Optional[float] = Field(None, ge=0.0, le=5.0)

class TherapistCreate(TherapistBase):
    pass

class TherapistUpdate(TherapistBase):
    email: Optional[EmailStr] = None
    license_number: Optional[str] = None
    license_state: Optional[str] = None
    is_active: Optional[bool] = None

class TherapistInDBBase(TherapistBase):
    id: UUID
    assessment_count: int
    is_active: bool
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class Therapist(TherapistInDBBase):
    pass

class TherapistInDB(TherapistInDBBase):
    pass